src/object_store.cpp

```diff
@@ -774,6 +774,37 @@ Schema ObjectStore::schema_from_group(Group const& group) {
     return schema;
 }
 
+util::Optional<Property> ObjectStore::property_for_column_index(ConstTableRef& table, size_t column_index)
+{
+    StringData column_name = table->get_column_name(column_index);
+
+#if REALM_HAVE_SYNC_STABLE_IDS
+    // The object ID column is an implementation detail, and is omitted from the schema.
+    // FIXME: Consider filtering out all column names starting with `__`.
```

Comment:
Would you mind updating this comment to say column names starting with `!`, since `sync::object_id_column_name` was changed to `!OID` at some point.